### PR TITLE
Simple and canonical name support for Structure Test Oracle files

### PR DESCRIPTION
**Pull request by niclasschuemann**
At the moment, ClassTest only allows simple class names for implemented interfaces and superclasses in the structure oracle test file. This is problematic because it is not validated that the actually correct interface or superclass is implemented.
If one exercise contains two interfaces with the same name but in different packages respectively, the ClassTest cannot determine which one of these is the correct one.
Therefore AJTS should be able to handle simple as well as canonical class names in the structure test oracle file correctly. 
Simple names can still be used if the location of an interface or a superclass is up to the student and canonical names will be used if the location of the interface or superclass is already given.

### CODE REVIEW

**Review by MaisiKoleni — CHANGES_REQUESTED** [review]
This is a good idea 👍 

Could you format the code (so that the code style check passes)? You can use Maven `spotless:apply` for that. And because of the second commit: the project here is using tabs, so it was correct before in that regard.

Regarding the changes: 
I think being able to compare the full canonical name of types is a good improvement in general. Could we use this approach consistently for classes in the structural tests? This would apply to e.g. attributes, parameter types and return types. Could you therefore create a protected method in `StructuralTestProvider` that is used for all such comparisons? This would also reduce the duplication of the logic.

**Comment by niclasschuemann** [comment]
I have implemented your suggested changes for almost all type occurrences except for attribute types since this is implemented a little bit differently. I will look into that tomorrow and formally request a re-review 👍

**Comment by MaisiKoleni** [comment]
> I have implemented your suggested changes for almost all type occurrences except for attribute types since this is implemented a little bit differently. I will look into that tomorrow and formally request a re-review 👍

We could also look into that together, I would have two more suggestions. You can contact me on the LS1-Slack.

File: src/main/java/de/tum/in/test/api/structural/StructuralTestProvider.java [hunk 1]
```diff
@@ -224,8 +229,8 @@ protected static boolean checkParameters(Class<?>[] observedParameters, JSONArra
 		}
 		/*
 		 * Create hash tables to store how often a parameter type occurs. Checking the
-		 * occurrences of a certain parameter type is enough, since the parameter names
-		 * or their order are not relevant to us.
+		 * occurrences of a certain parameter type is enough, since the parameter order
+		 * is not relevant to us.
 		 */
```
Comment (MaisiKoleni) [inline src/main/java/de/tum/in/test/api/structural/StructuralTestProvider.java:234]: For a followup: use the new type checking here as well.

**Review by MaisiKoleni — APPROVED** [review]
Looks good to me, parameters can be adjusted later as well. We could also think about a better way to generate canonical names, but this should be good enough for almost all cases.

**Pull request merged**